teams/infra/edit-and-sync-configurations-managed-by-argocd/action.yml

```diff
@@ -0,0 +1,121 @@
+  and pushed before an immediate sync is triggered (ArgoCD) to avoid waiting for the next periodic sync.
+
+  This GitHub action is specific to the technical stack and workflow of the Infra team, but the underlying actions upon which its
+  implementation is based can be reused and adapted to other workflows and needs.
+
+inputs:
+  argocd-app-name:
+    description: "ArgoCD application for which to trigger an immediate sync."
+    required: true
+  argocd-token:
+    description: "ArgoCD token with sufficient permissions to sync the ArgoCD application."
+    required: true
+  argocd-server:
+    description: |
+      URL of the ArgoCD server or one of the aliases 'dev', 'stage' or 'prod' which will be
+      automatically expanded with the URL of the corresponding Infra ArgoCD instance.
+    required: true
+  github-repository:
+    description: "The GitHub repository hosting the Kubernetes resource configurations to update."
+    required: true
+  github-repository-checkout-path:
+    description: "Relative path under $GITHUB_WORKSPACE to place the GitHUb repository"
+    default: "./repository"
+  github-repository-commit-author:
+    description: Value used for the commit author. Defaults to the username of whoever triggered this workflow run.
+    default: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
+  github-repository-commit-message:
+    description: "Value used for the commit message."
+  github-repository-branch:
+    description: "The branch to checkout."
+  github-token:
+    description: "A GitHub access token used to fetch the GitHub repository."
+    required: true
+  patches:
+    description: |
+      A list of YAML files with associated [YQ](https://github.com/mikefarah/yq)) expressions to apply in place (piped in the defined order).
+      Patches are applied using the underlying [yq-yaml-processor](../../../yq-yaml-processor) GitHub action.
+
+      See the example below:
+
+        patches: |
+          - file: path/to/a.yaml
+            expressions :
+            - '.a.count = 1'
+            - '.a.enabled = true'
+            - '.a.name = "A"'
+          - file: path/to/b.yaml
+            expressions :
+            - '.b.enabled = true'
+    required: true
+
+runs:
+  using: composite
+  steps:
+  - name: Checkout the target GitHub repository
+    uses: actions/checkout@v3
+    with:
+      path: ${{ inputs.github-repository-checkout-path }}
+      ref: ${{ inputs.github-repository-branch }}
+      repository: ${{ inputs.github-repository }}
+      token: ${{ inputs.github-token }}
+  - name: Prefix file paths in patches with the checkout path (if any)
+    uses: camunda/infra-global-github-actions/yq-yaml-processor@main
+    id: patches
+    env:
+      CHECKOUT_PATH: ${{ inputs.github-repository-checkout-path }}
+      PATCHES_TO_PREFIX_WITH_CHECKOUT_PATH: ${{ inputs.patches }}
+    with:
+      patches: |
+        - env: PATCHES_TO_PREFIX_WITH_CHECKOUT_PATH
```

Comment:
I don't understand how this works, could you elaborate?

Reply:
The var name is not really explicit. I can change it.

Reply:
This piece of code is not used anymore.